gcloud-java-compute/src/main/java/com/google/gcloud/compute/NetworkId.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Identity for a Google Compute Engine network.
+ */
+public final class NetworkId extends ResourceId {
+
+  private static final String REGEX = ResourceId.REGEX + "global/networks/([^/]+)";
+  private static final Pattern PATTERN = Pattern.compile(REGEX);
+  private static final long serialVersionUID = 2386765228138819506L;
+
+  private final String network;
+
+  NetworkId(String project, String network) {
+    super(project);
+    this.network = checkNotNull(network);
+  }
+
+  private NetworkId(NetworkId networkId) {
+    super(networkId.project());
+    this.network = checkNotNull(networkId.network());
+  }
+
+  /**
+   * Returns the name of the network. The network name must be 1-63 characters long and comply with
+   * RFC1035. Specifically, the name must match the regular expression
+   * {@code [a-z]([-a-z0-9]*[a-z0-9])?} which means the first character must be a lowercase letter,
+   * and all following characters must be a dash, lowercase letter, or digit, except the last
+   * character, which cannot be a dash.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  public final String network() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary final modifier in final class](https://www.codacy.com/app/mziccard/gcloud-java/file/2418453243/issues/source?bid=3140968&fileBranchId=3228978#l57)